sklearn/compose/_column_transformer.py

```diff
@@ -227,23 +237,25 @@ def _validate_transformers(self):
 
     def _validate_remainder(self, X):
         """Generate list of passthrough columns for 'remainder' case."""
-        if self.remainder not in ('drop', 'passthrough'):
+        is_transformer = ((hasattr(self.remainder, "fit")
+                           or hasattr(self.remainder, "fit_transform"))
+                          and hasattr(self.remainder, "transform"))
+        if (self.remainder not in ('drop', 'passthrough')
+                and not is_transformer):
             raise ValueError(
-                "The remainder keyword needs to be one of 'drop' or "
-                "'passthrough'. {0:r} was passed instead")
+                "The remainder keyword needs to be one of 'drop', "
+                "'passthrough', or estimator. '%s' was passed instead" %
+                self.remainder)
 
         n_columns = X.shape[1]
+        cols = []
+        for _, _, columns in self.transformers:
+            cols.extend(_get_column_indices(X, columns))
+        passthrough = sorted(list(set(range(n_columns)) - set(cols)))
+        passthrough = passthrough or None
 
-        if self.remainder == 'passthrough':
-            cols = []
-            for _, _, columns in self.transformers:
-                cols.extend(_get_column_indices(X, columns))
-            self._passthrough = sorted(list(set(range(n_columns)) - set(cols)))
-            if not self._passthrough:
-                # empty list -> no need to select passthrough columns
-                self._passthrough = None
-        else:
-            self._passthrough = None
+        self._remainder = ('remainder_transformer', self.remainder,
```

Comment:
just call it 'remainder'

Reply:
Does not pass `_validate_names` if this is named `remainder`. This is because `self.remainder` is defined in `__init__`.

Reply:
So then this needs to be excluded when invoking `_validate_names`

Reply:
I would also prefer to name it `remainder`.

Reply:
This is definitely not a use-case I'd foreseen for `_BaseComposition`, so if there's a more elegant way to structure `_BaseComposition`, feel free to change it!

Reply:
I think updating `_invalid_names` to let this through, would introduce an error when calling `set_params(remainder__{}={})` since there would be a `remainder` in transformer and a `self.remainder`.

Reply:
I will see what I can come up with for `_BaseComposition`.

Reply:
> I will see what I can come up with for _BaseComposition.

Don't go out of your way. Only if it seems justified by your needs.